pkg/apis/extensions/validation/validation.go

```diff
@@ -152,6 +165,8 @@ func ValidateThirdPartyResource(obj *extensions.ThirdPartyResource) field.ErrorL
 		version := &obj.Versions[ix]
 		if len(version.Name) == 0 {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("versions").Index(ix).Child("name"), version, "must not be empty"))
+		} else if !validation.IsDNS1123Label(version.Name) {
```

Comment:
yeah, I'm not sure what the right line to draw is. if there was a way for thirdpartyresources to specify what name formats they allow, that would be better, but in the absence of that, how much should we protect users of this API from bad names? see discussion at https://github.com/kubernetes/kubernetes/pull/25007#discussion-diff-61665610

> @liggitt: should we make the name validation tight (dns label or subdomain, etc) or minimal (anything we can represent as a URL path segment)?
>
> @deads2k: Without actual hooks to allow them to shape their names, I think forcing the label is reasonable.
> 
> @liggitt: you don't think most API consumers would be equipped to handle resources named İ ♥ \n?